boa/profiling.py

```diff
@@ -233,21 +257,41 @@ def __repr__(self):
 
 # cache gas_used for all computation (including children)
 def cache_gas_used_for_computation(contract, computation):
-    profile = contract.line_profile(computation)
     env = contract.env
+
+    def _recurse(computation):
+        # recursion for child computations
+        for _computation in computation.children:
+            child_contract = env.lookup_contract(_computation.msg.code_address)
+
+            if child_contract is None:
+                # for black box contracts, we don't profile the contract,
+                # but we recurse into the subcalls
+                _recurse(_computation)
+            else:
+                cache_gas_used_for_computation(child_contract, _computation)
+
+    if not getattr(contract, "_can_line_profile", False):
```

Comment:
we can use the normal getter now it's in the base evm contract

Reply:
i don't want to depend on everything always inheriting from BaseEvmContract. for instance, the user or a plugin inserts something else into the env contract space.